src/main/java/ee/buerokratt/ruuter/domain/steps/http/HttpStep.java

```diff
@@ -28,58 +28,58 @@
-            DefaultHttpService globalHttpExceptionService = ci.getProperties().getDefaultServiceInCaseOfException();
-            if (localHttpExceptionServiceExists()) {
-                localHttpExceptionService.executeHttpDefaultService(ci, resultName);
-            } else if (globalHttpExceptionServiceExists(globalHttpExceptionService)) {
-                globalHttpExceptionService.executeHttpDefaultService(ci, resultName);
+    public void handleFailedResult(DslInstance di) {
+        super.handleFailedResult(di);
+        if (isNotAllowedHttpStatusCode(di, ((HttpStepResult) di.getContext().get(resultName)).getResponse().getStatusCodeValue())) {
+            DefaultHttpDsl globalHttpExceptionDsl = di.getProperties().getDefaultDslInCaseOfException();
+            if (localHttpExceptionDslExists()) {
+                localHttpExceptionDsl.executeHttpDefaultDsl(di, resultName);
+            } else if (globalHttpExceptionDslExists(globalHttpExceptionDsl)) {
+                globalHttpExceptionDsl.executeHttpDefaultDsl(di, resultName);
             }
         }
     }
 
     @Override
-    protected void logStep(Long elapsedTime, ConfigurationInstance ci) {
-        ApplicationProperties properties = ci.getProperties();
-        MappingHelper mappingHelper = ci.getMappingHelper();
-        Integer responseStatus = ((HttpStepResult) ci.getContext().get(resultName)).getResponse().getStatusCodeValue();
-        String responseBody = mappingHelper.convertObjectToString(((HttpStepResult) ci.getContext().get(resultName)).getResponse().getBody());
+    protected void logStep(Long elapsedTime, DslInstance di) {
+        ApplicationProperties properties = di.getProperties();
+        MappingHelper mappingHelper = di.getMappingHelper();
+        Integer responseStatus = ((HttpStepResult) di.getContext().get(resultName)).getResponse().getStatusCodeValue();
+        String responseBody = mappingHelper.convertObjectToString(((HttpStepResult) di.getContext().get(resultName)).getResponse().getBody());
         String responseContent = responseBody != null && properties.getLogging().getDisplayResponseContent() ? responseBody : "-";
         String requestContent = args.getBody() != null && properties.getLogging().getDisplayRequestContent() ? args.getBody().toString() : "-";
-        LoggingUtils.logStep(log, this, ci.getRequestOrigin(), elapsedTime, args.getUrl(), requestContent, responseContent, String.valueOf(responseStatus));
+        LoggingUtils.logStep(log, this, di.getRequestOrigin(), elapsedTime, args.getUrl(), requestContent, responseContent, String.valueOf(responseStatus));
     }
 
-    private boolean isAllowedHttpStatusCode(ConfigurationInstance ci, Integer response) {
-        return ci.getProperties().getHttpCodesAllowList().isEmpty() || ci.getProperties().getHttpCodesAllowList().contains(response);
+    private boolean isNotAllowedHttpStatusCode(DslInstance di, Integer response) {
```

Comment:
"negative" functions are harder to read and understand than positive ones. Leave this as "isAllowed..." and simply reverse the result with `!` where you need to use the negated value.